src/builder.rs

```diff
@@ -275,7 +269,7 @@ impl AstarteBuilder {
     }
 
     /// build Astarte client, call this before `connect`
-    pub async fn build(&mut self) -> Result<(), AstarteBuilderError> {
+    pub async fn build(&mut self) -> Result<Self, AstarteBuilderError> {
```

Comment:
Does it make sense that `build` returns the builder again? I would expect this to return an `AstarteSdk` instead